pkg/cmd/task/task_ls.go

```diff
@@ -0,0 +1,131 @@
+package task
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/cli"
+	"github.com/dcos/dcos-core-cli/pkg/mesos"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+	lib "github.com/mesos/mesos-go/api/v1/lib"
+	"github.com/spf13/cobra"
+)
+
+func newCmdTaskLs(ctx api.Context) *cobra.Command {
+	// all and completed are useless but we keep them for retrocompatibility.
+	var all, completed, long bool
+	var path string
+
+	cmd := &cobra.Command{
+		Use:   "ls <task> [<path>]",
+		Short: "Print the list of files in the Mesos task sandbox",
+		Args:  cobra.RangeArgs(1, 2),
+		PreRun: func(cmd *cobra.Command, args []string) {
+			if len(args) == 2 {
+				path = args[1]
+			} else {
+				path = "."
+			}
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			mesosClient, err := mesos.NewClientWithContext(ctx)
+			if err != nil {
+				return err
+			}
+
+			allTasks, err := mesosClient.Tasks()
+			if err != nil {
+				return err
+			}
+
+			taskExpr := regexp.MustCompile(args[0])
+			var tasks []lib.Task
+			for _, t := range allTasks {
+				if strings.Contains(t.TaskID.Value, args[0]) || taskExpr.MatchString(t.TaskID.Value) {
+					tasks = append(tasks, t)
+				}
+			}
+
+			var containerParentIDs []string
+			var executorIDs []string
+			for _, t := range tasks {
+				if t.Name == "" {
+					return fmt.Errorf("unable to find task '%s'", args[0])
+				}
+
+				containerID := t.Statuses[len(t.Statuses)-1].ContainerStatus.GetContainerID()
+				if containerID.GetParent() == nil {
+					containerParentIDs = append(containerParentIDs, containerID.Value)
+				} else {
+					containerParentIDs = append(containerParentIDs, containerID.GetParent().Value)
+				}
+
+				if t.ExecutorID != nil {
+					executorIDs = append(executorIDs, (*t.ExecutorID).Value)
+				} else {
+					executorIDs = append(executorIDs, t.TaskID.Value)
+				}
+			}
+
+			var agentsPaths = map[string]map[string]string{}
+			client := mesos.NewClient(pluginutil.HTTPClient(""))
+			for _, t := range tasks {
+				if _, ok := agentsPaths[t.AgentID.Value]; !ok {
+					agentPaths, err := client.Debug(t.AgentID.Value)
+					if err != nil {
+						return err
+					}
+					agentsPaths[t.AgentID.Value] = agentPaths
+				}
+			}
+
+			for i, t := range tasks {
+				if len(tasks) > 1 {
+					fmt.Fprintln(ctx.Out(), "===> "+t.TaskID.Value+" <===")
+				}
+				taskPath := "/frameworks/" + t.FrameworkID.Value + "/executors/" + executorIDs[i] + "/runs/" + containerParentIDs[i]
+				for key, value := range agentsPaths[t.AgentID.Value] {
+					if strings.HasSuffix(key, taskPath) {
```

Comment:
Should it really be suffix and not prefix?

Reply:
Yes, I've manually tested and the function to use in order to get the same behavior as in the Python CLI is to use `HasSuffix`.